docs/Admin_HTTP_API.md

```diff
@@ -0,0 +1,295 @@
+```
+
+#### Response
+
+```
+HTTP 200 OK
+```
+
+```json
+{
+  "hostname": "vinxi-server",
+  "version": "0.1.0",
+  "runtime": "go1.6",
+  "platform": "darwin",
+  "cpus": 8,
+  "gorutines": 13,
+  "links": {
+    "catalog": "/catalog",
+    "instances": "/instances",
+    "manager": "/manager",
+    "plugins": "/plugins",
+    "scopes": "/scopes"
+  }
+}
+```
+
+### Global plugins
+
+#### List plugins
+
+```
+GET /plugins
+```
+
+##### Response
+
+```json
+[
+  {
+    "id": "UGEZGfua5S3U0nDS",
+    "name": "auth",
+    "description": "Authorization and authentication protection",
+    "config": {
+      "scheme": "Bearer",
+      "token": "s3cr3t"
+    }
+  }
+]
+```
+
+#### List plugins
+
+```
+GET /plugins
+```
+
+##### Response
+
+```json
+[
+  {
+    "id": "UGEZGfua5S3U0nDS",
+    "name": "auth",
+    "description": "Authorization and authentication protection",
+    "config": {
+      "scheme": "Bearer",
+      "token": "s3cr3t"
+    }
+  }
+]
+```
+
+#### Register plugin
+
+```
+GET /plugins/{id}
+```
+
+##### Request Body
+
+```json
+{
+  "name":"auth", 
+  "config": {
+    "token": "s3cr3t"
+  }
+}
+```
+
+##### Response
+
+```json
+{
+  "id": "UGEZGfua5S3U0nDS",
+  "name": "auth",
+  "description": "Authorization and authentication protection",
+  "config": {
+    "scheme": "Bearer",
+    "token": "s3cr3t"
+  }
+}
+```
+
+#### Get plugin
+
+```
+GET /plugins/{id}
+```
+
+##### Response
+
+```json
+{
+  "id": "UGEZGfua5S3U0nDS",
+  "name": "auth",
+  "description": "Authorization and authentication protection",
+  "config": {
+    "scheme": "Bearer",
+    "token": "s3cr3t"
+  }
+}
+```
+
+#### Delete plugin
+
+```
+DELETE /plugins/{id}
+```
+
+#### Response
+
+```
+HTTP 204 No Content
+```
+
+### Instances
+
+#### List instances
+
+```
+GET /instances
+```
+
+#### Response
+
+```
+HTTP 200 OK
+```
+
+```json
+[
+  {
+    "info": {
+      "id": "KamlJHrvH2owdzIG",
+      "name": "default",
+      "description": "This a default proxy",
+      "hostname": "vinxi-server",
+      "platform": "darwin",
+      "server": {
+        "port": 3100,
+        "readTimeout": 0,
+        "writeTimeout": 0,
+        "address": ""
+      }
+    },
+    "scopes": [
+      {
+        "id": "jd7CDZ48IsQ0RJqu",
+        "name": "custom",
+        "rules": [
+          {
+            "id": "MizprxflGgjeQe06",
+            "name": "path",
+            "description": "Matches HTTP request URL path againts a given path pattern",
+            "config": {
+              "path": "\/foo\/(.*)"
+            }
+          }
+        ],
+        "plugins": [
+          {
+            "id": "AsgTHxwUgUcg9oet",
+            "name": "forward",
+            "description": "Forward HTTP traffic to remote servers",
+            "config": {
+              "url": "http:\/\/httpbin.org"
+            }
+          }
+        ]
+      }
+    ]
+  }
+]
+```
+
+#### Get instance
+
+```
+GET /instances/{id}
+```
+
+#### Response
+
+```
+HTTP 200 OK
+```
+
+```json
+{
+  "info": {
```

Comment:
Wouldn't it be more relevant to have all these info at the top level and not under `info`?